### ISSUE TITLE
Handling complaints of email marked as spam

### ISSUE THREAD

**Issue by miabecker** [department-of-veterans-affairs/notification-api#381]
# Value Statement

**As** a VANotify admin
**I want to** receive a notification
**When** an email is marked as spam by an end user 
**So that** I can see how many emails are being marked as spam

# Acceptance Criteria

**GIVEN** an email is marked by spam by an end user
**WHEN** there is a complaint sent back to our email provider
**AND** our email providers passes the complaint back to us
**THEN** a notification to the VANotify team as defined by our VANotify service

# Checklist
- [x] Create a celery task for sending complaint notification
- [x] Call the celery task when receiving complaint (in parallel to complaint callback)
- [x] Define static template in database and configuration, similar to other templates internal to VANotify
- [x] Send emails to team members of VANotify service

# Assumptions
- VANotify service exists and team members are added to it

# Additional Info/Resources
- Can be tested using AWS email simulator - https://docs.aws.amazon.com/ses/latest/DeveloperGuide/send-email-simulator.html

# Out of Scope
- Dashboard view

# Open Questions
- Does Granicus do this as well or just Amazon SES?  SES only, Granicus (kind of but there is separate story) >> avoid being AWS specific
- Considering two options:
  - (1) new task to call send_notification endpoint (POST request)
  - (2) call on deliver_email task (SEND_EMAIL queue) after creating complaint notification
- What metrics do we want to capture as part of this new addition?
  - number of complaints (graph)
  - no need for alert
- What are high level attributes to capture in template?
  - see `service_callback_tasks#create_complaint_callback_data` as reference for desired info
  - service name, template name, notification id
  - resource: complaints in admin UI
- Where or how should we define the recipient address, the va-notify group?
  - store as config (va-notifications@thoughtworks.com)
  - vanotify service users

**Comment by saman-moshafi-va** [comment]
The branch is `origin/381-notify-email-complaints`
We prefer to use the va-notify service which is maintained by migrations and its id is defined in config as NOTIFY_SERVICE_ID.  One potential issue is that the notify service is always linked to the default user, `Notify service user` with email, `va-notify-user@public.govdelivery.com`.  Should the complaints notification sent to this email as well?

**Comment by lingtran** [comment]
![Screen Shot 2021-03-31 at 2.43.33 PM.png](https://images.zenhubusercontent.com/5ecea5442d18e437e3b23d00/b521ccb9-6d5d-4d0e-9f90-317d041b61ff)

**Comment by TWs-DevenS** [comment]
This ticket has been reviewed. Acceptance criteria has been met. Moving the ticket to signoff.

**Comment by miabecker** [comment]
@lingtran has this been enabled in production yet?

**Comment by TWs-DevenS** [comment]
**QA Verification**
Send email to user account, 
As user  mark email as spam
System should process this action as a complaint and sent back to our email provider
AND our email providers passes the complaint back to us
Verified complaint count (see scr shot above)  which verifies a notification is sent to the VANotify team as defined by our VANotify service
